src/logger/logger.hpp

```diff
@@ -1,9 +1,15 @@
 #pragma once
 #include "file_manager.hpp"
-
 #include <AsyncLogger/Logger.hpp>
 using namespace al;
 
+namespace stream
+{
+	inline auto net_events   = std::make_shared<LogStream>("net_events");
```

Comment:
I can bet there is no good reason for this to be shared and not just unique_ptr and the logger itself just have them as non owning pointers (aka just raw pointers) but since this is related to an upstream dependency now I guess it can stay like that

Reply:
That's a fair point, and we should probably use unique_ptr since there's only one instance (basically making a singleton pattern instance by inlining it, which is just a very roundabout way of doing static).

That being said, it should be static unique_ptr instead of inline.

Reply:
global static inline variables in hpp is the equivalent of creating hidden bugs.
Static for global variables inside hpp files means that the constructor is called everytime the file is included somewhere, c++ moment.